variables.tf

```diff
@@ -276,6 +276,12 @@ variable "deployment_minimum_healthy_percent" {
   default     = 100
 }
 
+variable "availability_zone_rebalancing" {
+  type        = string
+  description = "ECS automatically redistributes tasks within a service across Availability Zones (AZs) to mitigate the risk of impaired application availability due to underlying infrastructure failures and task lifecycle activities. The valid values are `ENABLED` and `DISABLED`."
+  default     = null
```

Comment:
Mind keeping the validation to allow null, ENABLED, and DISABLED?

Please also add to the description that the default is `DISABLED` when the input is set to `null`

Reply:
it doesn't make sense to use `null` for this attribute cause it `DISABLED` by default
i've tried to add `availability_zone_rebalancing` for your example with `null` or `DISABLED` and it's created with `DISABLED` value

![Screenshot 2024-12-04 at 11 51 45](https://github.com/user-attachments/assets/da957f86-c50b-4a6e-9792-5d8944c143d7)

i can add ugly hack for variable validation block with `null` but it will not give any sense

```
variable "availability_zone_rebalancing" {
  type        = string
  description = "ECS automatically redistributes tasks within a service across Availability Zones (AZs) to mitigate the risk of impaired application availability due to underlying infrastructure failures and task lifecycle activities. The valid values are `ENABLED` and `DISABLED`."
  default = null

  validation {
    condition     = (var.availability_zone_rebalancing == null ? true : contains(["ENABLED", "DISABLED"], var.availability_zone_rebalancing))
    error_message = "The valid values are `ENABLED` and `DISABLED`."
  }
}
```

https://github.com/hashicorp/terraform-provider-aws/blob/v5.77.0/internal/service/ecs/service.go#L557-L562

reverting to previous commit

Reply:
Oh i see, thanks for trying